### PR TITLE
github-ci: enable Fedora 33 packaging

### LINKED ISSUE #5502: Fedora 33 released but in repo available for 31  only
Tarantool version: 2.4

OS version: Fedora 32 and Fedora 33

Bug description: repo not exist 

Steps to reproduce: try install repo on Fedora 32 and 33
```sh
 curl -L https://tarantool.io/ByZVhS/release/2.4/installer.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   184  100   184    0     0    696      0 --:--:-- --:--:-- --:--:--   699
100  9357  100  9357    0     0  11259      0 --:--:-- --:--:-- --:--:--     0
Detected operating system as fedora/32.
Unfortunately, your operating system is not supported by this script.
```

If try manual 
```sh
dnf -y install sudo
sudo rm -f /etc/yum.repos.d/*tarantool*.repo
sudo tee /etc/yum.repos.d/tarantool_2_4.repo <<- EOF
[tarantool_2_4]
name=Fedora-\$releasever - Tarantool
baseurl=https://download.tarantool.org/tarantool/release/2.4/fedora/\$releasever/x86_64/
gpgkey=https://download.tarantool.org/tarantool/release/2.4/gpgkey
repo_gpgcheck=1
gpgcheck=0
enabled=1

[tarantool_2_4-source]
name=Fedora-\$releasever - Tarantool Sources
baseurl=https://download.tarantool.org/tarantool/release/2.4/fedora/\$releasever/SRPMS
gpgkey=https://download.tarantool.org/tarantool/release/2.4/gpgkey
repo_gpgcheck=1
gpgcheck=0
EOF

sudo dnf -q makecache -y --disablerepo='*' --enablerepo='tarantool_2_4'
sudo dnf -y install tarantool
- Status code: 404 for https://hb.bizmrg.com/tarantool_repo/release/2.4/fedora/32/x86_64/repodata/repomd.xml (IP: 95.163.144.221)
Error: Не удалось загрузить метаданные для репозитория «tarantool_2_4»: Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried


```

### PR DESCRIPTION
**Pull request by avtikhon**
1.    github-ci: add Fedora 33 to packaging
    
    Added jobs for testing and deploying Fedora 33 packages.
    
    Added Fedora 33 at update_repo tool to make able to save packages
    in S3 buckets.
    
    Closes #5502

2. build: enable packaging for Fedora 33
    
    Since Python 2 was deprecated then new OS do not have repositories
    with its packages. It is not good way to create all these packages
    manually and store it as backported. Let's wait when Python 3 will
    be used for testing and for now the testing disabled for Fedora 33
    packaging.
    
    Also found that %cmake macros used in build spec was changed - new
    option was added:
    
      -B x86_64-redhat-linux-gnu
    
    It changed the build path and broke the build, to fix it these
    options was additionally locally reset to:
    
      -B .
    
    Found that LTO was set by default in Fedora 33, check
    
      /usr/lib/rpm/redhat/macros
    
        # LTO is the default in Fedora.
        #   "%define _lto_cflags %{nil}"  to opt out
        #
        # We currently have -ffat-lto-objects turned on out of an abundance of
        # caution.  To remove it we need to do a check of the installed .o/.a files
        # to verify they have real sections/symbols after LTO stripping.  That
        # way we can detect installing an unusable .o/.a file.  This is on the TODO
        # list for F34.
        %_gcc_lto_cflags        -flto=auto -ffat-lto-objects
        %_clang_lto_cflags      -flto
        %_lto_cflags            %{expand:%%{_%{toolchain}_lto_cflags}}
    
        %_general_options       -O2 %{?_lto_cflags} -fexceptions -g -grecord-gcc-switches -pipe
    
    and build issues occured:
    
      /build/usr/src/debug/tarantool-2.8.0.2/src/lib/core/reflection.h:124:33: warning: type of â€˜METHODS_SENTINELâ€™ does not match original declaration [-Wlto-type-mismatch]
        124 | extern const struct method_info METHODS_SENTINEL;
            |                                 ^
      /build/usr/src/debug/tarantool-2.8.0.2/src/lib/core/reflection.c:35:26: note: â€˜METHODS_SENTINELâ€™ was previously declared here
         35 | const struct method_info METHODS_SENTINEL = {
            |                          ^
    
    it was fixed as was suggested in 1st way of [1][2]. This warning was
    disabled in commit:
    
      f9e28ce4602aff3f9bb4e743b0d6167b0f8df88d ("Add LTO support")
    
    with cmake flag:
    
      -D_ENABLE_LTO=ON
    
    so added this flag to cmake RPM spec build flags for Fedora 33 and its
    later versions.
    
    Needed for #5502
    Needed for #5697
    
    [1] - https://github.com/tarantool/tarantool/issues/5697#issuecomment-759409465
    [2] - https://github.com/tarantool/tarantool/pull/5707#discussion_r561072044

### CODE REVIEW

File: rpm/tarantool.spec [hunk 1]
```diff
@@ -76,6 +76,11 @@ BuildRequires: libunwind-devel
 %endif
 
 # Set dependences for tests.
+# Since Python 2 was depricated then new OS do not have repositories
```
Comment (Totktonada) [inline rpm/tarantool.spec:79]: Typo: depricated → deprecated (in the commit message too).

File: rpm/tarantool.spec [hunk 2]
```diff
@@ -159,6 +167,9 @@ C and Lua/C modules.
          -DWITH_SYSTEMD:BOOL=ON \
          -DSYSTEMD_UNIT_DIR:PATH=%{_unitdir} \
          -DSYSTEMD_TMPFILES_DIR:PATH=%{_tmpfilesdir} \
+%endif
+%if 0%{?fedora} >= 33
+         -DENABLE_LTO=ON \
```
Comment (Totktonada) [inline rpm/tarantool.spec:172]: In fact, we set our own flags `-flto -fno-fat-lto-objects` (via `/usr/share/cmake/Modules/Compiler/GNU.cmake`) (and `-Wno-lto-type-mismatch` for a linker) after [ones](https://fedoraproject.org/wiki/LTOByDefault) from Fedora 33 (`%_lto_cflags` == `-flto=auto -ffat-lto-objects`). Fedora 35 [will](https://fedoraproject.org/wiki/Changes/LTOBuildImprovements) enable `-fno-fat-lto-objects` by default, because there is no much sense to store compiled code in object/archive files if you don't install them (and it saves some build time). According to GCC [docs](https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html) `-flto` should make attempt to detect GNU Make job server (as well as `-flto=auto`) if I understood the following wording right:

> You can also specify `-flto=jobserver` to use GNU make’s job server mode to determine the number of parallel jobs. \<...\> Even without the option value, GCC tries to automatically detect a running GNU make’s job server.

So, to sum up, the change looks okay for me. We don't discard anything important from the distribution specific build options.

File: rpm/tarantool.spec [hunk 3]
```diff
@@ -147,7 +153,9 @@ C and Lua/C modules.
 
 %build
 # RHBZ #1301720: SYSCONFDIR an LOCALSTATEDIR must be specified explicitly
-%cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+# since Fedora 33 LTO flags were set in CC/LD flags by default
```
[the remark this answers is not in the document]
Comment (igormunkin) [reply rpm/tarantool.spec:156]: > It is better to place this comment right before the flag.

I believe this change will separate the multiline cmake command by parts and break it. I've tried to do this in scope of [this](https://lists.tarantool.org/tarantool-patches/cover.1611149713.git.imun@tarantool.org/T/#t) series but failed.

[the remark this answers is not in the document]
Comment (Totktonada) [reply rpm/tarantool.spec:156]: @igormunkin Thanks for the clarification.

@avtikhon So, please, fix punctuation. I would also suggest to separate different comments with an 'empty' line:

```
# Here I described a point A.
#
# Here I describe another point, B.
```

[the remark this answers is not in the document]
Comment (avtikhon) [reply rpm/tarantool.spec:156]: Not possible to set the commit just before the flag, because of syntax. Anyway changed the message to point the flags:
```
-# since Fedora 33 LTO flags were set in CC/LD flags by default
-%cmake -S . -B . \
+# Must specified explicitly since Fedora 33:
+# 1. -B .
+#    because for now binary path by default value like:
+#      '-B x86_64-redhat-linux-gnu'
+# 2. -DENABLE_LTO=ON
+#    because for now LTO flags are set in CC/LD flags by default:
+#      '-flto=auto -ffat-lto-objects'
+%cmake -B . \
```

[the remark this answers is not in the document]
Comment (Totktonada) [reply rpm/tarantool.spec]: Now it is here.

**Review by Totktonada — APPROVED** [review]
I see no downsides. Let's rebase and push after PR #5716.

File: rpm/tarantool.spec [hunk 4]
```diff
@@ -76,6 +76,11 @@ BuildRequires: libunwind-devel
 %endif
 
 # Set dependences for tests.
+# Since Python 2 was deprecated then new OS do not have repositories
```
Comment (Totktonada) [inline rpm/tarantool.spec:79]: Nit: I would add 'build' prefix for the spec fix.

**Review by kyukhin — APPROVED** [review]
